### PR TITLE
Recipe schema org thrift

### PR DESCRIPTION
**Pull request by frederickobrien**
The first step in finally getting recipe schema onto the Guardian website and reaping the SEO benefits. This PR adds a Thrift definition for Schema.org recipes.

Companions downstream: 

- https://github.com/guardian/content-api-scala-client/pull/408
- https://github.com/guardian/content-api/pull/2858
- https://github.com/guardian/facia-scala-client/pull/310
- https://github.com/guardian/frontend/pull/26931

For more information on the fields included see https://schema.org/Recipe.

Preview deployed via https://github.com/guardian/gha-scala-library-release-workflow/blob/main/docs/making-a-release.md#choose-release-type

### CODE REVIEW

File: models/src/main/thrift/content/v1.thrift [hunk 1]
```diff
@@ -1683,6 +1684,10 @@ struct ContentChannel {
 
 }
 
+struct SchemaOrg {
+    1: optional list<schemaorg.SchemaRecipe> recipe
+}
+
```
Comment (emdash-ie) [inline models/src/main/thrift/content/v1.thrift:1690]: Is this better here or in `schemaorg.thrift`? I’m not sure, would appreciate your thoughts.

On the one hand, assuming we add more schemas here in future, if we ever want to reuse this definition in other places it’s probably nicer to have it in `schemaorg.thrift`.

On the other, maybe we don’t want it to be reused anywhere, and it makes more sense to stay here.

File: models/src/main/thrift/content/schemaorg.thrift
Comment (emdash-ie) [inline models/src/main/thrift/content/schemaorg.thrift:1]: I find the file name `schemaorg.thrift` slightly hard to read. It’s a pretty minor preference, but what would you say to `schema_org.thrift`? (My ideal would be `schema-org.thrift`, but that’s an entirely new style, whereas we already have `contentatom.thrift` and `story_package_article.thrift`.)

**Review by emdash-ie — APPROVED** [review]
This looks good to me, but I’d appreciate delaying merging to discuss the (minor) points in my comments, if that’s ok?

File: models/src/main/thrift/content/schemaorg.thrift
Comment (frederickobrien) [reply models/src/main/thrift/content/schemaorg.thrift:1]: I'm partial to a dash myself but feel consistency is probably the best approach here. Have changed with [`45423ba` (#237)](https://github.com/guardian/content-api-models/pull/237/commits/45423ba8ff1935aff22fd207ce460b422033aed1)

Comment (emdash-ie) [reply models/src/main/thrift/content/schemaorg.thrift:1]: (I like dashes mainly because they don’t require me to press shift to type them)

**Pull request merged**